homeassistant/components/fritzbox/binary_sensor.py

```diff
@@ -1,74 +1,52 @@
 """Support for Fritzbox binary sensors."""
-import requests
-
 from homeassistant.components.binary_sensor import (
     DEVICE_CLASS_WINDOW,
     BinarySensorEntity,
 )
-from homeassistant.const import CONF_DEVICES
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_ENTITY_ID,
+    ATTR_NAME,
+    ATTR_UNIT_OF_MEASUREMENT,
+)
+from homeassistant.core import HomeAssistant
 
-from .const import CONF_CONNECTIONS, DOMAIN as FRITZBOX_DOMAIN, LOGGER
+from . import FritzBoxEntity
+from .const import CONF_COORDINATOR, DOMAIN as FRITZBOX_DOMAIN
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
-    """Set up the Fritzbox binary sensor from config_entry."""
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Set up the Fritzbox binary sensor from ConfigEntry."""
     entities = []
-    devices = hass.data[FRITZBOX_DOMAIN][CONF_DEVICES]
-    fritz = hass.data[FRITZBOX_DOMAIN][CONF_CONNECTIONS][config_entry.entry_id]
-
-    for device in await hass.async_add_executor_job(fritz.get_devices):
-        if device.has_alarm and device.ain not in devices:
-            entities.append(FritzboxBinarySensor(device, fritz))
-            devices.add(device.ain)
-
-    async_add_entities(entities, True)
-
-
-class FritzboxBinarySensor(BinarySensorEntity):
+    coordinator = hass.data[FRITZBOX_DOMAIN][entry.entry_id][CONF_COORDINATOR]
+
+    for ain, device in coordinator.data.items():
+        if device.has_alarm:
```

Comment:
```suggestion
        if not device.has_alarm:
            return
```

Then outdent the block below